joint_teapot/config.py

```diff
@@ -19,6 +19,12 @@ class Settings(BaseSettings):
     # git
     repos_dir: str = "./repos"
 
+    # mattermost
+    mattermost_url: str = "focs.ji.sjtu.edu.cn"
```

Comment:
consider renaming it as `focs_url` and replacing other occurrences of the URL in the code

Reply:
These are just the default settings, I think it's good to leave some room for future flexibility.

Reply:
The variable name is not reasonable since the domain is focs.ji.sjtu.edu.cn and the URL is focs.ji.sjtu.edu.cn/mm

Reply:
It happens to be the case that the mm server and git server share the same domain name with focs, and we have to thank our helpful and competent IT for this. Generally this is an anti-pattern and we should avoid making such assumptions.

Reply:
It happens to be the case that the mm server and git server share the same domain name with focs, and we have to thank our helpful and competent IT for this. Generally this is an anti-pattern and we should avoid making such assumptions.

Reply:
so we need these settings: `gitea_domain`, `gitea_suffix`, `mattermost_domain`, `mattermost_suffix`, `canvas_domain`

Reply:
Will add `canvas_domain` later. Others have already been implemented.